deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -877,12 +896,21 @@ def _update_persist_config(self, ds_config):
         Init.param_persistence_threshold = ds_config.zero_config.param_persistence_threshold
         Init.model_persistence_threshold = ds_config.zero_config.model_persistence_threshold // self.num_partitions
 
+    def _zero_init_param(self, param):
+        self._convert_to_deepspeed_param(param)
```

Comment:
Do we need this conditional branch? 
When `if dist.get_world_group() == self.get_dp_process_group()`, `dist.get_global_rank(self.get_dp_process_group(), 0)` returns 0.
The "else" branch seems to cover another branch.

Reply:
The `if` branch covers when we are using only data parallelism. The else branch covers the case of tensor parallelism (or pipeline parallelism) where the rank 0 of each tensor slice may not be the global rank 0.